evm/src/main/java/org/hyperledger/besu/evm/operation/AuthCallOperation.java

```diff
@@ -0,0 +1,100 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.evm.operation;
+
+import static org.hyperledger.besu.evm.internal.Words.clampedToLong;
+
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.internal.Words;
+
+/** Introduced via EIP-3074 to call another contract with a different authorization context. */
+public class AuthCallOperation extends AbstractCallOperation {
+
+  /**
+   * Instantiates a new AuthCallOperation.
+   *
+   * @param gasCalculator a Prague or later gas calculator
+   */
+  public AuthCallOperation(final GasCalculator gasCalculator) {
+    super(0xF7, "AUTHCALL", 7, 1, gasCalculator);
+  }
+
+  @Override
+  protected Address to(final MessageFrame frame) {
+    return Words.toAddress(frame.getStackItem(1));
+  }
+
+  @Override
+  protected Wei value(final MessageFrame frame) {
+    return Wei.wrap(frame.getStackItem(2));
+  }
+
+  @Override
+  protected Wei apparentValue(final MessageFrame frame) {
+    return value(frame);
+  }
+
+  @Override
+  protected long inputDataOffset(final MessageFrame frame) {
+    return clampedToLong(frame.getStackItem(3));
+  }
+
+  @Override
+  protected long inputDataLength(final MessageFrame frame) {
+    return clampedToLong(frame.getStackItem(4));
+  }
+
+  @Override
+  protected long outputDataOffset(final MessageFrame frame) {
+    return clampedToLong(frame.getStackItem(5));
+  }
+
+  @Override
+  protected long outputDataLength(final MessageFrame frame) {
+    return clampedToLong(frame.getStackItem(6));
+  }
+
+  @Override
+  protected Address address(final MessageFrame frame) {
+    return to(frame);
+  }
+
+  @Override
+  protected Address sender(final MessageFrame frame) {
+    return frame.getAuthorizedBy();
+  }
+
+  @Override
+  public long gasAvailableForChildCall(final MessageFrame frame) {
+    return gasCalculator().gasAvailableForChildCall(frame, gas(frame), !value(frame).isZero());
+  }
+
+  @Override
+  public OperationResult execute(final MessageFrame frame, final EVM evm) {
+    if (frame.isStatic() && !value(frame).isZero()) {
+      return new OperationResult(cost(frame, true), ExceptionalHaltReason.ILLEGAL_STATE_CHANGE);
+    } else if (frame.getAuthorizedBy() != null) {
+      return super.execute(frame, evm);
+    } else {
+      return new OperationResult(cost(frame, true), null);
```

Comment:
Failures should push `frame.pushStackItem(LEGACY_FAILURE_STACK_ITEM);` (i.e. zero) to the stack prior to returning an `OperationResult`